frontend/src/component/changeRequest/ChangeRequest/Changes/Change/VariantPatch/Diff.tsx

```diff
@@ -17,13 +17,17 @@ interface IDiffProps {
     data: any;
 }
 
+const variantsArrayToObject = (arr: any) =>
+    arr.reduce((a: any, v: any) => ({ ...a, [v.name]: v }), {});
```

Comment:
Same question regarding any: is it something we're avoiding or is it fine?

If we don't want anys, how about this (also a couple of parameter naming suggestions)? 

```suggestion
const variantsArrayToObject = (variants: { name: string }[]) =>
    variants.reduce((acc, { name, ...rest }) => ({ ...acc, [name]: rest }), {});
```

Would the linter be happy with eliding the types in the lambda? Because otherwise it gets kind of verbose 🙈 

```suggestion
const variantsArrayToObject = (variants: { name: string }[]) =>
    variants.reduce(
        (acc: { [name: string]: {} }, { name, ...rest }: { name: string }) => ({
            ...acc,
            [name]: rest,
        }),
        {},
    );
```

Reply:
Yep, all great suggestions. Addressed in https://github.com/Unleash/unleash/pull/3307/commits/4e802032923a3e557411516547617897f7998784.